connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -362,13 +362,15 @@ public int executePut(String url, String body) throws IOException {
         try (OutputStreamWriter out = new OutputStreamWriter(httpCon.getOutputStream())) {
             out.write(body);
         }
-        try (InputStream is = httpCon.getInputStream()) {
-            int c;
-            StringBuilder response = new StringBuilder();
-            while ((c = is.read()) != -1) {
-                response.append((char) c);
+        if (httpCon.getResponseCode() == 200 || httpCon.getResponseCode() == 201) {
```

Comment:
This is done because of the fact that `HttpUrlConnection` throws `IOException` when we try to process the `InputStream` for error responses. We aren't really doing anything with the response other than logging it. So, I thought it would be reasonable to not process the stream for error scenarios.

Reply:
My concern is that, because this is intended to be used in connector projects, we should also be handling the error response. What if we do the following (which is similar to something you suggested offline):
```
InputStream _is;
if (httpConn.getResponseCode() < HttpURLConnection.HTTP_BAD_REQUEST) {
    try (InputStream is = httpConn.getInputStream()) {
        log.info("Put response for URL={} is {}", url, responseToString(is));
    }
} else {
    try (InputStream is = httpConn.getErrorStream()) {
        log.info("Put error response for URL={} is {}", url, responseToString(is));
    }
}
```
where
```
private String responseToString(InputStream stream) throws IOException {
    int c;
    StringBuilder response = new StringBuilder();
    while ((c = is.read()) != -1) {
        response.append((char) c);
    }
    return response.toString();
}
```
WDYT?